applications/GeoMechanicsApplication/tests/cpp_tests/test_time_stepping.cpp

```diff
@@ -0,0 +1,154 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//                   Anne van de Graaf
+//
+
+#include "testing/testing.h"
+#include "custom_workflows/time_step_executor.h"
+#include "custom_workflows/strategy_wrapper.hpp"
+#include "solving_strategies/strategies/solving_strategy.h"
+
+using namespace Kratos;
+
+
+namespace
+{
+
+class ProcessSpy : public Process
+{
+public:
+    void ExecuteInitializeSolutionStep() override {mSolutionStepInitializedCalls += 1;}
+    void ExecuteFinalizeSolutionStep()   override {mSolutionStepFinalizedCalls   += 1;}
+
+    unsigned int NumberOfExecuteInitializeSolutionStepCalls() const {return mSolutionStepInitializedCalls;}
+    unsigned int NumberOfExecuteFinalizeSolutionStepCalls()   const {return mSolutionStepFinalizedCalls;}
+
+private:
+    unsigned int mSolutionStepInitializedCalls = 0;
+    unsigned int mSolutionStepFinalizedCalls   = 0;
+};
+
+class DummyStrategyWrapper : public StrategyWrapper
+{
+public:
+    explicit DummyStrategyWrapper::DummyStrategyWrapper(bool convergence_needed) :
+        mConvergenceNeeded ( convergence_needed ? TimeStepEndState::ConvergenceState::converged : TimeStepEndState::ConvergenceState::non_converged) {}
```

Comment:
Instead of using a `bool`ean  flag, we could also have passed the desired convergence state directly (i.e. `TimeStepEndState::ConvergenceState::converged` or `TimeStepEndState::ConvergenceState::non_converged`). This may result in slightly more readable constructor calls: `DummyStrategyWrapper{true}` vs `DummyStrategyWrapper{TimeStepEndState::ConvergenceState::converged}` (where we may want to define an alias to reduce the latter to `DummyStrategyWrapper{converged}`. What do you think?

Reply:
Done ( I hope as intended ... )